### DISCUSSION TITLE
Customised DVCs for 2 way communication between client and server

### DISCUSSION THREAD

**Discussion by dp2608** [FreeRDP/FreeRDP#7030] [General]
Hello

I wanted to create a customized DVC for two way communication between client and server so that during every session required data can be exchanged. Since I am new to FreeRDP, I was kind of lost. It would be of great help if someone can provide references to a step by step guide to creating a DVC. 

Thank you.

**Comment by akallabeth** [comment]
check https://github.com/FreeRDP/FreeRDP/tree/master/channels/echo for a (minimal) channel implementation.
you can build these outside of FreeRDP source tree too, you just need to start with `/dvc:mychannelname` (having the `libmychannelname.so` in freerdp plugin folder)
